src/lib.rs

```diff
@@ -610,24 +633,25 @@ where
 /// # Examples
 ///
 /// ```
-/// use textwrap::unfill;
+/// use textwrap::{LineEnding, unfill};
 ///
 /// let (text, options) = unfill("\
 /// * This is an
 ///   example of
 ///   a list item.
-/// ");
+/// ", LineEnding::LF);
 ///
 /// assert_eq!(text, "This is an example of a list item.\n");
 /// assert_eq!(options.initial_indent, "* ");
 /// assert_eq!(options.subsequent_indent, "  ");
 /// ```
-pub fn unfill(text: &str) -> (String, Options<'_>) {
-    let trimmed = text.trim_end_matches('\n');
+pub fn unfill(text: &str, line_ending: LineEnding) -> (String, Options<'_>) {
+    let line_ending_pat = line_ending.as_str();
+    let trimmed = text.trim_end_matches(line_ending_pat);
     let prefix_chars: &[_] = &[' ', '-', '+', '*', '>', '#', '/'];
 
-    let mut options = Options::new(0);
-    for (idx, line) in trimmed.split('\n').enumerate() {
+    let mut options = Options::new(0).line_ending(line_ending);
+    for (idx, line) in trimmed.split(line_ending_pat).enumerate() {
```

Comment:
What about auto-detecting the line ending here? Basically, if we find a `"\r\n"` in `text` then we use that line ending, otherwise we use `"\n"`.

I think it would be okay to say that `unfill` doesn't handle input with a mix of `\r\n` and `\n` line endings (if the caller is concerned about such cases, I would suggest the caller to normalize the input first).

Reply:
I gave it a try and after some tinkering I made it work with mixed input and also made it 20% faster.

```
original ❯ cargo criterion --bench "unfill"
   Compiling textwrap v0.15.0 (/home/.../develop/textwrap)
    Finished bench [optimized] target(s) in 4.54s
  Executable benches/unfill.rs (target/release/deps/unfill-d9aa0a94ff4a34d4)
unfill                  time:   [19.092 us 19.105 us 19.117 us]                    
                        change: [+25.614% +25.731% +25.857%] (p = 0.00 < 0.05)
                        Performance has regressed.
updated ❯ cargo criterion --bench "unfill"
   Compiling textwrap v0.15.0 (/home/koiuo/develop/textwrap)
    Finished bench [optimized] target(s) in 4.57s
  Executable benches/unfill.rs (target/release/deps/unfill-d9aa0a94ff4a34d4)
unfill                  time:   [15.246 us 15.255 us 15.266 us]                    
                        change: [-19.845% -19.726% -19.585%] (p = 0.00 < 0.05)
                        Performance has improved.
```

I haven't updated a single existing test (except for added assertions) and haven't touched the prefix detection code. But I feel like we could make `unfill` work with multiple input paragraphs without significant code complication or performance degradation. If there's an interest I could give it a try after we are done with this PR.